[bug] Flaky `install-all` pre-submit test

The `install-all` test fails with the following message about 75% of the time. It seems that aqua is trying to modify the `slsa-verifier` binary while it's being executed. If the test is re-run a few times it succeeds.

```
Run make clean all
  make clean all
  shell: /usr/bin/bash -e {0}
  env:
    GH_TOKEN: ***
/tmp/tmp.ocPb14l6yI.tar.gz: OK
Verified build using builder "https://github.com/slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@refs/tags/v[2](https://github.com/ianlewis/dotfiles/actions/runs/15624686304/job/44016588464#step:4:2).1.0" at commit a8c17eb9e[3](https://github.com/ianlewis/dotfiles/actions/runs/15624686304/job/44016588464#step:4:3)5245bdf6fc9505e75c86d225a870[4](https://github.com/ianlewis/dotfiles/actions/runs/15624686304/job/44016588464#step:4:4)9
Verifying artifact /tmp/tmp.JMRHGT483w.aqua-v2.[5](https://github.com/ianlewis/dotfiles/actions/runs/15624686304/job/44016588464#step:4:5)3.1.tar.gz: PASSED

PASSED: SLSA verification passed
time=2025-0[6](https://github.com/ianlewis/dotfiles/actions/runs/15624686304/job/44016588464#step:4:7)-13T01:58:26Z level=info msg=download and unarchive the package env=linux/amd64 package_name=aqua-proxy package_version=v1.2.9 program=aqua program_version=2.53.1 registry=
time=2025-06-13T01:58:2[7](https://github.com/ianlewis/dotfiles/actions/runs/15624686304/job/44016588464#step:4:8)Z level=info msg=create a symbolic link env=linux/amd64 package_name=aqua-proxy package_version=v1.2.9 program=aqua program_version=2.53.1 registry=
time=2025-06-13T01:5[8](https://github.com/ianlewis/dotfiles/actions/runs/15624686304/job/44016588464#step:4:9):27Z level=info msg=create a symbolic link command=shellcheck env=linux/amd64 package_name=koalaman/shellcheck package_version=v0.10.0 program=aqua program_version=2.53.1
time=2025-06-13T01:58:27Z level=info msg=create a symbolic link command=actionlint env=linux/amd64 package_name=rhysd/actionlint package_version=v1.7.7 program=aqua program_version=2.53.1
time=2025-06-13T01:58:27Z level=info msg=create a symbolic link command=golangci-lint env=linux/amd64 package_name=golangci/golangci-lint package_version=v1.64.8 program=aqua program_version=2.53.1
time=2025-06-13T01:58:27Z level=info msg=create a symbolic link command=shfmt env=linux/amd64 package_name=mvdan/sh package_version=v3.11.0 program=aqua program_version=2.53.1
time=2025-06-13T01:58:27Z level=info msg=create a symbolic link command=selene env=linux/amd64 package_name=Kampfkarren/selene package_version=0.28.0 program=aqua program_version=2.53.1
time=2025-06-13T01:58:27Z level=info msg=create a symbolic link command=stylua env=linux/amd64 package_name=JohnnyMorganz/StyLua package_version=v2.0.2 program=aqua program_version=2.53.1
time=2025-06-13T01:58:27Z level=info msg=create a symbolic link command=rg env=linux/amd64 package_name=BurntSushi/ripgrep package_version=14.1.1 program=aqua program_version=2.53.1
time=2025-06-13T01:58:27Z level=info msg=create a symbolic link command=tofu env=linux/amd64 package_name=opentofu/opentofu package_version=v1.[9](https://github.com/ianlewis/dotfiles/actions/runs/15624686304/job/44016588464#step:4:10).0 program=aqua program_version=2.53.1
time=2025-06-13T01:58:27Z level=info msg=create a symbolic link command=fzf env=linux/amd64 package_name=junegunn/fzf package_version=v0.62.0 program=aqua program_version=2.53.1
time=2025-06-13T01:58:27Z level=info msg=create a symbolic link command=gh env=linux/amd64 package_name=cli/cli package_version=v2.70.0 program=aqua program_version=2.53.1
time=2025-06-13T01:58:27Z level=info msg=create a symbolic link command=todos env=linux/amd64 package_name=ianlewis/todos package_version=v0.13.0 program=aqua program_version=2.53.1
time=2025-06-13T01:58:27Z level=info msg=create a symbolic link command=slsa-verifier env=linux/amd64 package_name=slsa-framework/slsa-verifier package_version=v2.7.0 program=aqua program_version=2.53.1
time=2025-06-13T01:58:27Z level=info msg=create a symbolic link command=cosign env=linux/amd64 package_name=sigstore/cosign package_version=v2.5.0 program=aqua program_version=2.53.1
time=2025-06-13T01:58:27Z level=info msg=download and unarchive the package env=linux/amd64 package_name=Kampfkarren/selene package_version=0.28.0 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:27Z level=info msg=download and unarchive the package env=linux/amd64 package_name=koalaman/shellcheck package_version=v0.[10](https://github.com/ianlewis/dotfiles/actions/runs/15624686304/job/44016588464#step:4:11).0 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:27Z level=info msg=download and unarchive the package env=linux/amd64 package_name=golangci/golangci-lint package_version=v1.64.8 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:27Z level=info msg=download and unarchive the package env=linux/amd64 package_name=mvdan/sh package_version=v3.[11](https://github.com/ianlewis/dotfiles/actions/runs/15624686304/job/44016588464#step:4:12).0 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:27Z level=info msg=download and unarchive the package env=linux/amd64 package_name=rhysd/actionlint package_version=v1.7.7 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:28Z level=info msg=download and unarchive the package env=linux/amd64 package_name=JohnnyMorganz/StyLua package_version=v2.0.2 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:28Z level=info msg=download and unarchive the package env=linux/amd64 package_name=BurntSushi/ripgrep package_version=14.1.1 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:28Z level=info msg=download and unarchive the package env=linux/amd64 package_name=opentofu/opentofu package_version=v1.9.0 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:28Z level=info msg=download and unarchive the package env=linux/amd64 package_name=junegunn/fzf package_version=v0.62.0 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:28Z level=info msg=download and unarchive the package env=linux/amd64 package_name=cli/cli package_version=v2.70.0 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:28Z level=info msg=download and unarchive the package env=linux/amd64 package_name=ianlewis/todos package_version=v0.13.0 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:28Z level=info msg=download and unarchive the package env=linux/amd64 package_name=slsa-framework/slsa-verifier package_version=v2.7.0 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:28Z level=info msg=download and unarchive the package env=linux/amd64 package_name=sigstore/cosign package_version=v2.5.0 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:28Z level=info msg=verify a package with slsa-verifier env=linux/amd64 package_name=ianlewis/todos package_version=v0.13.0 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:28Z level=info msg=download and unarchive the package env=linux/amd64 package_name=slsa-framework/slsa-verifier package_version=v2.7.0 program=aqua program_version=2.53.1 registry=
time=2025-06-13T01:58:29Z level=info msg=verify a package with slsa-verifier env=linux/amd64 package_name=slsa-framework/slsa-verifier package_version=v2.7.0 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:29Z level=info msg=verify GitHub Artifact Attestations env=linux/amd64 package_name=cli/cli package_version=v2.70.0 program=aqua program_version=2.53.1 registry=standard
time=2025-06-13T01:58:29Z level=info msg=download and unarchive the package env=linux/amd64 package_name=cli/cli package_version=v2.74.0 program=aqua program_version=2.53.1 registry=
Verified signature against tlog entry index 169679096 at URL: https://rekor.sigstore.dev/api/v1/log/entries/108e9186e8c5677af62eb[12](https://github.com/ianlewis/dotfiles/actions/runs/15624686304/job/44016588464#step:4:13)d269d15ef29fe221e7b73a2ed1eccba714b967f27632dc2c511ed0a5f
Verified build using builder "https://github.com/slsa-framework/slsa-github-generator/.github/workflows/builder_go_slsa3.yml@refs/tags/v2.0.0" at commit 6657aada084353c65e5dde35394b1a010289fab0
Verifying artifact /tmp/060523960: PASSED

PASSED: SLSA verification passed
time=2025-06-[13](https://github.com/ianlewis/dotfiles/actions/runs/15624686304/job/44016588464#step:4:14)T01:58:30Z level=error msg=install the package env=linux/amd64 error=open the file (/home/runner/.local/share/aquaproj-aqua/pkgs/github_release/github.com/slsa-framework/slsa-verifier/v2.7.0/slsa-verifier-linux-amd64/slsa-verifier-linux-amd64): open /home/runner/.local/share/aquaproj-aqua/pkgs/github_release/github.com/slsa-framework/slsa-verifier/v2.7.0/slsa-verifier-linux-amd64/slsa-verifier-linux-amd64: text file busy package_name=slsa-framework/slsa-verifier package_version=v2.7.0 program=aqua program_version=2.53.1 registry=standard
Verified build using builder "https://github.com/slsa-framework/slsa-github-generator/.github/workflows/builder_go_slsa3.yml@refs/tags/v2.1.0" at commit c541b34b863a2c82d59da3cd50658d7d636ff39a
Verifying artifact /tmp/084938365: PASSED

PASSED: SLSA verification passed
time=2025-06-13T01:58:33Z level=fatal msg=aqua failed env=linux/amd64 error=it failed to install some packages program=aqua program_version=2.53.1
make: *** [Makefile:662: install-aqua] Error 1
Error: Process completed with exit code 2.
```

--- a/.github/workflows/pull_request.tests.yml
+++ b/.github/workflows/pull_request.tests.yml
@@ -49,10 +49,10 @@ jobs:
         with:
           node-version-file: ".node-version"
 
-      - name: Install all
+      - name: Install
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: make clean all
+        run: make clean install
 
   # Test that a clean install works on Linux arm64.
   install-all-ubuntu2404-arm64:
@@ -68,10 +68,10 @@ jobs:
         with:
           node-version-file: ".node-version"
 
-      - name: Install all
+      - name: Install
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: make clean all
+        run: make clean install
 
   # Test that a clean install works on MacOS.
   install-all-macos26-arm64:
@@ -99,10 +99,10 @@ jobs:
             coreutils \
             libyaml
 
-      - name: Install all
+      - name: Install
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: gmake clean all
+        run: gmake clean install
 
   # NOTE: needed for protected branch checks.
   install-all:

--- a/Makefile
+++ b/Makefile
@@ -39,6 +39,17 @@ OUTPUT_FORMAT ?= $(shell if [ "${GITHUB_ACTIONS}" == "true" ]; then echo "github
 REPO_ROOT := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
 REPO_NAME := $(shell basename "$(REPO_ROOT)")
 
+
+# TODO(github.com/aquaproj/aqua/issues/3951): workaround for flaky aqua install
+# renovate: datasource=github-releases depName=slsa-framework/slsa-verifier versioning=loose
+SLSA_VERIFIER_VERSION ?= v2.7.1
+SLSA_VERIFIER_REPO := github.com/slsa-framework/slsa-verifier
+SLSA_VERIFIER_CHECKSUM.linux.amd64 := 946DBEC729094195E88EF78E1734324A27869F03E2C6BD2F61CBC06BD5350339
+SLSA_VERIFIER_CHECKSUM.linux.arm64 := 5D3B2349EDE7BFEC19E7A21569F18B9F7410145AD12E9584B175370669E14061
+SLSA_VERIFIER_CHECKSUM.darwin.arm64 := 39ABFCF5F1D690C3E889CE3D2D6A8B87711424D83368511868D414E8F8BCB05C
+SLSA_VERIFIER_CHECKSUM ?= $(SLSA_VERIFIER_CHECKSUM.$(kernel).$(arch))
+SLSA_VERIFIER_URL := https://$(SLSA_VERIFIER_REPO)/releases/download/$(SLSA_VERIFIER_VERSION)/slsa-verifier-$(kernel)-$(arch)
+
 # renovate: datasource=github-releases depName=aquaproj/aqua versioning=loose
 AQUA_VERSION ?= v2.55.1
 AQUA_REPO := github.com/aquaproj/aqua
@@ -213,18 +224,19 @@ $(HOME)/opt:
 #####################################################################
 
 .PHONY: all
-all: install-all configure-all ## Install and configure everything.
+all: test install-all ## Run all tests, install and configure everything.
 
-.PHONY: configure-all
-configure-all: configure-aqua configure-bash configure-bat configure-efm-langserver configure-ghostty configure-git configure-nix configure-nvim configure-tmux ## Configure all tools.
+.PHONY: install
+install: install-tools install-runtimes configure ## Install and configure everything.
 
-.PHONY: install-all
-install-all: install-tools install-runtimes ## Install all CLI tools and runtimes.
+.PHONY: configure
+configure: configure-aqua configure-bash configure-bat configure-efm-langserver configure-ghostty configure-git configure-nix configure-nvim configure-tmux
 
 .PHONY: install-tools
-install-tools: install-bin install-aqua ## Install all CLI tools.
+install-tools: install-bin install-slsa-verifier install-aqua
 
-install-runtimes: install-go install-node install-python install-ruby ## Install all runtimes.
+.PHONY: install-runtimes
+install-runtimes: install-go install-node install-python install-ruby
 
 ## Testing
 #####################################################################
@@ -843,6 +855,17 @@ configure-tmux: ## Configure tmux.
 
 ## Install Tools
 #####################################################################
+
+$(XDG_BIN_HOME)/slsa-verifier: $(XDG_BIN_HOME)
+	@# bash \
+	tempfile=$$($(MKTEMP) --suffix=".slsa-verifier-$(SLSA_VERIFIER_VERSION)"); \
+	curl -sSLo "$${tempfile}" "$(SLSA_VERIFIER_URL)"; \
+	echo "$(SLSA_VERIFIER_CHECKSUM)  $${tempfile}" | shasum -a 256 -c; \
+	chmod +x "$${tempfile}"; \
+	mv "$${tempfile}" $@
+
+.PHONY: install-slsa-verifier
+install-slsa-verifier: $(XDG_BIN_HOME)/slsa-verifier ## Install slsa-verifier
 
 # NOTE: The go runtime is required to install some tools on some platforms.
 .PHONY: install-aqua

--- a/aqua/aqua-checksums.json
+++ b/aqua/aqua-checksums.json
@@ -2236,66 +2236,6 @@
       "algorithm": "sha256"
     },
     {
-      "id": "github_release/github.com/slsa-framework/slsa-verifier/v2.7.0/slsa-verifier-darwin-amd64",
-      "checksum": "36694B43AB23BE234ADD09272E5FAF77349D7E267BF65C01DC9BCDF58C4F496E",
-      "algorithm": "sha256"
-    },
-    {
-      "id": "github_release/github.com/slsa-framework/slsa-verifier/v2.7.0/slsa-verifier-darwin-arm64",
-      "checksum": "84D9122CE12E0C79080844285FD5C4976407ED3463E434A1B21B0979C46B1E55",
-      "algorithm": "sha256"
-    },
-    {
-      "id": "github_release/github.com/slsa-framework/slsa-verifier/v2.7.0/slsa-verifier-linux-amd64",
-      "checksum": "499BEFB675EFCCA9001AFE6E5156891B91E71F9C07AB120A8943979F85CC82E6",
-      "algorithm": "sha256"
-    },
-    {
-      "id": "github_release/github.com/slsa-framework/slsa-verifier/v2.7.0/slsa-verifier-linux-arm64",
-      "checksum": "DC3845D7605F666A0938389C1C5735230E50B32A547867FFD351FB14DF928167",
-      "algorithm": "sha256"
-    },
-    {
-      "id": "github_release/github.com/slsa-framework/slsa-verifier/v2.7.0/slsa-verifier-windows-amd64.exe",
-      "checksum": "61FF8B1CCA6AC0012B0BA906367836F64A389444766BE437DF2A69F71285F43B",
-      "algorithm": "sha256"
-    },
-    {
-      "id": "github_release/github.com/slsa-framework/slsa-verifier/v2.7.0/slsa-verifier-windows-arm64.exe",
-      "checksum": "DDF58798049599C44CAF299B6A9CF8A41760DAA94EE208BDAE8AA78FC75DCB2B",
-      "algorithm": "sha256"
-    },
-    {
-      "id": "github_release/github.com/slsa-framework/slsa-verifier/v2.7.1/slsa-verifier-darwin-amd64",
-      "checksum": "4BAF25415727821F847A38BCCEDC86C3E5B17CBFC2EB534CD554FEB6C856D6F1",
-      "algorithm": "sha256"
-    },
-    {
-      "id": "github_release/github.com/slsa-framework/slsa-verifier/v2.7.1/slsa-verifier-darwin-arm64",
-      "checksum": "39ABFCF5F1D690C3E889CE3D2D6A8B87711424D83368511868D414E8F8BCB05C",
-      "algorithm": "sha256"
-    },
-    {
-      "id": "github_release/github.com/slsa-framework/slsa-verifier/v2.7.1/slsa-verifier-linux-amd64",
-      "checksum": "946DBEC729094195E88EF78E1734324A27869F03E2C6BD2F61CBC06BD5350339",
-      "algorithm": "sha256"
-    },
-    {
-      "id": "github_release/github.com/slsa-framework/slsa-verifier/v2.7.1/slsa-verifier-linux-arm64",
-      "checksum": "5D3B2349EDE7BFEC19E7A21569F18B9F7410145AD12E9584B175370669E14061",
-      "algorithm": "sha256"
-    },
-    {
-      "id": "github_release/github.com/slsa-framework/slsa-verifier/v2.7.1/slsa-verifier-windows-amd64.exe",
-      "checksum": "1D8F61AD747ECC3D375D2A563CEBF2991748B7DA1A9BDA9A500804C3C499E3C0",
-      "algorithm": "sha256"
-    },
-    {
-      "id": "github_release/github.com/slsa-framework/slsa-verifier/v2.7.1/slsa-verifier-windows-arm64.exe",
-      "checksum": "44144E98328D221F0490EF6B4A58A465DEFE8F697F387ABBBF07EF5ADB68D4AC",
-      "algorithm": "sha256"
-    },
-    {
       "id": "github_release/github.com/sxyazi/yazi/v25.5.31/yazi-aarch64-apple-darwin.zip",
       "checksum": "8BDF137E8F84CD11F9DD866FF2B68E4886D1055AB21FE2E0C6F6A0515D2BFD70",
       "algorithm": "sha256"

--- a/aqua/aqua.yaml
+++ b/aqua/aqua.yaml
@@ -36,7 +36,6 @@ packages:
   - name: junegunn/fzf@v0.66.0
   - name: cli/cli@v2.82.0
   - name: ianlewis/todos@v0.13.0
-  - name: slsa-framework/slsa-verifier@v2.7.1
   - name: sigstore/cosign@v2.6.1
   - name: go-acme/lego@v4.26.0
   - name: astral-sh/ruff@0.14.0

